reference schema name when creating index

kgeis opened the pull request:
fixes #2

kgeis committed:
Update index.sql

fixes #2

brylie reviewed (changes requested):
Please update the README to document this feature.

kgeis commented:
This is a bug fix, not a feature. I don't understand how it would fit in the README.

brylie commented:
It looks like this is an undocumented feature. Would you mind adding some documentation to the README?

brylie commented:
@drewbanin can we please get a review/approval for this bug fix?

brylie commented on macros/optimizers/index.sql:
```diff
@@ -1,5 +1,5 @@
 {% macro index(this, column) %}
 
-create index if not exists "{{ this.name }}__index_on_{{ column }}" on {{ this }} ("{{ column }}")
```
Is it common to prefix the index name with `schema_name.`?

brylie replied:
I.e. the index name should already be isolated to the schema, correct? If so, we probably don't need to include the schema in the index name.

drewbanin commented:
hey @jtcohen6 - do you think you could take a look at this one?

brylie commented:
It's also worth considering the impact native dbt Postgres index creation support will have on this package.

https://github.com/fishtown-analytics/dbt/issues/804

jtcohen6 commented:
Right on — I just merged https://github.com/fishtown-analytics/dbt/issues/3106, which adds native support for Postgres indexes in `dbt-postgres`. The tact we took there was, rather than namespacing indexes to specific schemas, creating a globally unique (hashed) name for each index based on the relation, properties, and creation timestamp.

If either of you think that's a mistake—that indexes _should_ be namespaced to the same schema as the relation they're defined on—we can still change that before releasing v0.20.0.

In any case, I'm happy to approve and merge this PR as is, while also acknowledging that we should announce the deprecation of this package.

brylie commented:
I think the approach you took is fine, as long as the index names will still be human-readable and relatively concise. Agreed on deprecating this package unless it may serve as a foundation for other Postgres-specific functionality.

jtcohen6 commented:
@brylie The approach over in fishtown-analytics/dbt#3106 prioritizes global uniqueness and conciseness over human readability. (The index name is just a MD5 hash.) Could you say a bit more about your desire for human readability? It makes sense intuitively, and I want to make sure we're taking a balanced approach to these trade-offs.

kgeis committed:
remove schema qualifier from index name

From PostgreSQL docs: "No schema name can be included here; the index is always created in the same schema as its parent table."

kgeis commented:
It looks like you have a different approach, but I still updated my PR for correctness. It was not appropriate for me to reference a schema in the index name.
